File: src/java/org/javaturk/dp/ch05/flyweight/book/Book.java
package org.javaturk.dp.ch05.flyweight.book;

import java.util.ArrayList;
import java.util.List;

import org.javaturk.dp.ch04.builder.computer4.Computer;
import org.javaturk.dp.ch04.builder.computer4.ComputerBuilderI;
import org.javaturk.dp.ch04.builder.computer4.Computer.ComputerBuilder;

public class Book {
	private String bookName;
	private List<Page> pages;
	private static Character endOfPage = new Character(12);

	public Book(String bookName, int numberOfPages) {
		this.bookName = bookName;
		pages = new ArrayList<>(numberOfPages);
	}

	public void add(Page page) {
		pages.add(page);
	}

	public List<Page> getPages() {
		return pages;
	}

	public String toString() {
		StringBuilder sb = new StringBuilder();
		for (Page page : pages) {
			sb.append(page.toString());
			sb.append(endOfPage.toString());
		}
		return sb.toString();
	}
}

File: src/typescript/com/gokselkucuksahin/dp/structural-patterns/flyweight/book/book.ts
import {Page} from "./page";
import {Character} from "./character";

export class Book {
  private bookName: string;
  private pages: Page[];
  private static endOfPage: Character = Character.character_1(String.fromCharCode(12));

  constructor(bookName: string, numberOfPages: number) {
    this.bookName = bookName;
    this.pages = Array<Page>(numberOfPages);
  }

  add(page: Page): void {
    this.pages.push(page);
  }

  getPages(): Page[] {
    return this.pages;
  }

  toString(): string {
    const str: string[] = [];
    this.pages.forEach(page => {
      str.push(page.toString());
      str.push(Book.endOfPage.toString());
    })
    return str.join('');
  }
}